visearch-android/src/main/java/com/visenze/visearch/android/ProductImageSearchParams.java

```diff
@@ -0,0 +1,101 @@
+package com.visenze.visearch.android;
+
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+import com.visenze.visearch.android.model.Image;
+
+public class ProductImageSearchParams extends ProductBaseSearchParams {
+
+    @SerializedName("im_url")
+    private String imUrl;
+
+    @SerializedName("im_id")
+    private String imId;
+
+    @SerializedName("image")
+    @Expose(deserialize = false, serialize = false)
+    private Image image;
+
+    @SerializedName("box")
+    private int[] box;
+
+    @SerializedName("detection")
+    private String detection;
+
+    @SerializedName("detection_limit")
+    private Integer detectionLimit;
+
+    @SerializedName("detection_sensitivity")
+    private String detectionSensitivity;
+
+    public String getImUrl() {
+        return imUrl;
+    }
+
+    public void setImUrl(String imUrl) {
+        this.imUrl = imUrl;
+    }
+
+    public String getImId() {
+        return imId;
+    }
+
+    public void setImId(String imId) {
+        this.imId = imId;
+    }
+
+    public Image getImage() {
+        return image;
+    }
+
+    public void setImage(Image image) {
+        this.image = image;
+    }
+
+    public int[] getBox() {
+        return box;
+    }
+
+    public void setBox(int[] box) {
```

Comment:
can we have a wrapper method to setBox as (x1, y1,x2, y2) to prevent users from sending an invalid box array

Reply:
or setBox(Box box)

Reply:
changed already

Reply:
invalid, check my comment